tuna/utils/function.py

```diff
@@ -0,0 +1,394 @@
+# pylint: disable=attribute-defined-outside-init
```

Comment:
please do not use this flag, set your attributes=None inside the init so pylint is happy. This is what we do in rest of Tuna

Reply:
defining an attribute as `None` has different semantics to not defining it at all (not just in theory, but the code is actually relying on this fact in some places)